build-plugin/common-dependencies/src/org/jetbrains/kotlinx/jupyter/common/httpUtil.kt

```diff
@@ -22,18 +22,18 @@ class ResponseWrapper(
     val url: String,
 ) : Response by response
 
-fun createHttpClient(): HttpHandler {
+private fun createHttpClient(): HttpHandler {
     return ClientFilters.FollowRedirects().then(ApacheClient())
 }
 
-fun httpRequest(request: Request): ResponseWrapper {
+fun <R> httpRequest(request: Request, mapper: (ResponseWrapper) -> R): R {
     val client = createHttpClient()
     val response = client(request)
 
-    return ResponseWrapper(response, request.uri.toString())
+    return ResponseWrapper(response, request.uri.toString()).use(mapper)
```

Comment:
Won't it fix the issue if we just close the response here? I mean
```
val client = createHttpClient()
val response = client(request)
response.close()
return ResponseWrapper(response, request.uri.toString())
```

Reply:
The connection will be closed of course, but I'm afraid that if you try to access the request's body after that, you'll get an exception

Reply:
Actually, it seems that closing the response doesn't help at this point. The returned response object is a MemoryResponse, so closing it is a no-op. The HTTP connection is probably expected to be closed in `CloseableHttpResponse.toHttp4kResponse` which consumes the entire payload, but again, that doesn't happen in our AWS setup.

Reply:
@strangepleasures have you checked that the edits you have introduced fix the issue in your AWS setup?

Reply:
I can provide you with pip package if you need

Reply:
`ApacheClient()` without arguments creates a client with a `PoolingHttpClientConnectionManager` with infinite TTL. Maybe, that's why connections never get closed.